src/core_ct/core.py

```diff
@@ -282,6 +343,30 @@ def chunk(self, x1=0, y1=0, z1=0, x2=None, y2=None, z2=None) -> Core:
         new_core = Core(self.pixel_array[x1:x2, y1:y2, z1:z2], self.pixel_dimensions)
         return new_core
 
+
+    def filter(self, brightness_filter) -> Core:
+        """
+        Get a three-dimensional section of the core scan.
+
+        Arguments:
+        ---------
+            brightness_filter: lambda function that defines what will be filtered out.
+                               This function must either return the passed value or set
+                               it to None.
+
+        Returns:
+        -------
+            New core object with only the specified brightness values left.
+        """
+        core_filtered = self.pixel_array.copy()
+        for i, row in enumerate(self.pixel_array):
+            for j, col in enumerate(row):
+                for k, brightness in enumerate(col):
+                    core_filtered[i][j][k] = brightness_filter(brightness)
```

Comment:
Typically a filter lambda returns a bool, this is more of a `map` function

Maybe we should keep this code and have a `map` function as well, but for the `filter` function I think the lambda should return a bool

Reply:
I agree with @TheSecondReal0 on making the lambda return a `bool`. Ideally, the user should just pass a boolean lambda function, and then within the function we handle setting filtered-out values to `np.nan` or whatever null value we are using. 

Also, we should look to see if there are faster ways of running the lambda on the whole array. I think this Stack Overflow post might have some good ideas, specifically in regards to using `np.vectorize`: https://stackoverflow.com/questions/35215161/most-efficient-way-to-map-function-over-numpy-array